econml/metalearners/_metalearners.py

```diff
@@ -170,6 +171,7 @@ def fit(self, Y, T, X=None, *, inference=None):
             X = np.zeros((Y.shape[0], 1))
         Y, T, X, _ = check_inputs(Y, T, X, multi_output_T=False)
         T = self._one_hot_encoder.fit_transform(T.reshape(-1, 1))
+        self._set_encoded_treatment_names(self._one_hot_encoder, drop_first=True)
```

Comment:
I think the fact that this single line appears everywhere is a bit problematic and the reason why I think this logic should be going into the TreatmentExpansionMixin. For instance I believe you've forgotten to add this line to DRIV and more generally, this could become even hairier to maintain with more and more estimators.